### ISSUE TITLE
List analysis page generating error

### ISSUE THREAD

**Issue by julie-sullivan** [intermine/intermine#1449]
http://beta.flymine.org/beta/bagDetails.do?scope=all&bagName=FlyMine_Alzheimers-Use-Case
```
   
The following gives more detail. This stacktrace is only visible to the admin user
      java.lang.NullPointerException: Path: &quot;Gene.secondaryIdentifier&quot;, pathToIndex: &quot;{}&quot;, prefix: &quot;Gene&quot;, query: &quot;<query name=&quot;&quot; model=&quot;genomic&quot; view=&quot;Gene.secondaryIdentifier Gene.symbol Gene.primaryIdentifier Gene.organism.name&quot; longDescription=&quot;&quot;><constraint path=&quot;Gene&quot; op=&quot;IN&quot; value=&quot;FlyMine_Alzheimers-Use-Case&quot;/></query>&quot;
	at org.intermine.api.results.WebResults.translateRow(WebResults.java:396)
	at org.intermine.api.results.WebResults.getResultElements(WebResults.java:375)
	at org.intermine.web.logic.results.PagedTable.updateRows(PagedTable.java:773)
	at org.intermine.web.logic.results.PagedTable.setPageAndPageSize(PagedTable.java:252)
	at org.intermine.web.struts.BagDetailsController.execute(BagDetailsController.java:240)
	at org.apache.struts.tiles.actions.TilesAction.execute(TilesAction.java:76)
	at org.apache.struts.action.RequestProcessor.processActionPerform(RequestProcessor.java:425)
	at org.apache.struts.action.RequestProcessor.process(RequestProcessor.java:228)
	at org.apache.struts.action.ActionServlet.process(ActionServlet.java:1913)
	at org.apache.struts.action.ActionServlet.doGet(ActionServlet.java:449)
	at javax.servlet.http.HttpServlet.service(HttpServlet.java:622)
	at javax.servlet.http.HttpServlet.service(HttpServlet.java:729)
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:291)
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:206)
	at org.apache.catalina.core.ApplicationDispatcher.invoke(ApplicationDispatcher.java:720)
	at org.apache.catalina.core.ApplicationDispatcher.doInclude(ApplicationDispatcher.java:584)
	at org.apache.catalina.core.ApplicationDispatcher.include(ApplicationDispatcher.java:523)
	at org.apache.struts.tiles.UrlController.execute(UrlController.java:92)
	at org.apache.struts.tiles.taglib.InsertTag$InsertHandler.doEndTag(InsertTag.java:879)
	at org.apache.struts.tiles.taglib.InsertTag.doEndTag(InsertTag.java:465)
	at org.apache.jsp.layout_jsp._jspx_meth_tiles_005fget_005f1(layout_jsp.java:1112)
	at org.apache.jsp.layout_jsp._jspService(layout_jsp.java:278)
	at org.apache.jasper.runtime.HttpJspBase.service(HttpJspBase.java:70)
	at javax.servlet.http.HttpServlet.service(HttpServlet.java:729)
	at org.apache.jasper.servlet.JspServletWrapper.service(JspServletWrapper.java:438)
	at org.apache.jasper.servlet.JspServlet.serviceJspFile(JspServlet.java:396)
	at org.apache.jasper.servlet.JspServlet.service(JspServlet.java:340)
	at javax.servlet.http.HttpServlet.service(HttpServlet.java:729)
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:291)
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:206)
	at org.apache.tomcat.websocket.server.WsFilter.doFilter(WsFilter.java:52)
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:239)
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:206)
	at org.apache.catalina.core.ApplicationDispatcher.invoke(ApplicationDispatcher.java:720)
	at org.apache.catalina.core.ApplicationDispatcher.processRequest(ApplicationDispatcher.java:466)
	at org.apache.catalina.core.ApplicationDispatcher.doForward(ApplicationDispatcher.java:391)
	at org.apache.catalina.core.ApplicationDispatcher.forward(ApplicationDispatcher.java:318)
	at org.apache.struts.action.RequestProcessor.doForward(RequestProcessor.java:1083)
	at org.apache.struts.tiles.TilesRequestProcessor.doForward(TilesRequestProcessor.java:295)
	at org.apache.struts.tiles.TilesRequestProcessor.processTilesDefinition(TilesRequestProcessor.java:271)
	at org.apache.struts.tiles.TilesRequestProcessor.internalModuleRelativeForward(TilesRequestProcessor.java:370)
	at org.apache.struts.action.RequestProcessor.processForward(RequestProcessor.java:566)
	at org.apache.struts.action.RequestProcessor.process(RequestProcessor.java:212)
	at org.apache.struts.action.ActionServlet.process(ActionServlet.java:1913)
	at org.apache.struts.action.ActionServlet.doGet(ActionServlet.java:449)
	at javax.servlet.http.HttpServlet.service(HttpServlet.java:622)
	at javax.servlet.http.HttpServlet.service(HttpServlet.java:729)
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:291)
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:206)
	at org.apache.tomcat.websocket.server.WsFilter.doFilter(WsFilter.java:52)
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:239)
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:206)
	at org.intermine.web.filters.SetCharacterEncodingFilter.doFilter(SetCharacterEncodingFilter.java:139)
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:239)
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:206)
	at org.apache.catalina.core.StandardWrapperValve.invoke(StandardWrapperValve.java:217)
	at org.apache.catalina.core.StandardContextValve.invoke(StandardContextValve.java:106)
	at org.apache.catalina.core.StandardHostValve.invoke(StandardHostValve.java:142)
	at org.apache.catalina.valves.ErrorReportValve.invoke(ErrorReportValve.java:79)
	at org.apache.catalina.valves.AbstractAccessLogValve.invoke(AbstractAccessLogValve.java:616)
	at org.apache.catalina.core.StandardEngineValve.invoke(StandardEngineValve.java:88)
	at org.apache.catalina.connector.CoyoteAdapter.service(CoyoteAdapter.java:518)
	at org.apache.coyote.ajp.AbstractAjpProcessor.process(AbstractAjpProcessor.java:844)
	at org.apache.coyote.AbstractProtocol$AbstractConnectionHandler.process(AbstractProtocol.java:673)
	at org.apache.tomcat.util.net.NioEndpoint$SocketProcessor.doRun(NioEndpoint.java:1500)
	at org.apache.tomcat.util.net.NioEndpoint$SocketProcessor.run(NioEndpoint.java:1456)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1145)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:615)
	at org.apache.tomcat.util.threads.TaskThread$WrappingRunnable.run(TaskThread.java:61)
	at java.lang.Thread.run(Thread.java:745)
```

**Comment by julie-sullivan** [comment]
an error in the log, not sure if related?

```
bag tables: 0 ms, generate: 0 ms, optimise: 1 ms,  ms,  estimate: 1 ms, execute: 4 ms, convert results: 63 ms, extra queries: 14 ms, total: 83 ms, rows: 193
110830982 [ajp-bio-8009-exec-11051] ERROR org.intermine.api.profile.StorableBag  - Exception committing bagValues for bag: 59000002
java.sql.SQLException: Error writing to database, running statement COPY bagvalues (savedbagid, value, extra) FROM STDIN BINARY, data size = 7418
        at org.intermine.sql.writebatch.FlushJobPostgresCopyImpl.flush(FlushJobPostgresCopyImpl.java:56)
        at org.intermine.sql.writebatch.Batch$BatchFlusher.run(Batch.java:461)
        at java.lang.Thread.run(Thread.java:745)
Caused by: org.postgresql.util.PSQLException: ERROR: duplicate key value violates unique constraint "bagvalues_index1"
  Detail: Key (savedbagid, value, extra)=(59000002, A1AT_HUMAN, H. sapiens) already exists.
  Where: COPY bagvalues, line 1
        at org.postgresql.core.v3.QueryExecutorImpl.receiveErrorResponse(QueryExecutorImpl.java:2161)
        at org.postgresql.core.v3.QueryExecutorImpl.processCopyResults(QueryExecutorImpl.java:966)
        at org.postgresql.core.v3.QueryExecutorImpl.endCopy(QueryExecutorImpl.java:828)
        at org.postgresql.core.v3.CopyInImpl.endCopy(CopyInImpl.java:59)
        at org.postgresql.copy.CopyManager.copyIn(CopyManager.java:181)
        at org.postgresql.copy.CopyManager.copyIn(CopyManager.java:161)
        at org.intermine.sql.writebatch.FlushJobPostgresCopyImpl.flush(FlushJobPostgresCopyImpl.java:51)
        ... 2 more
110830984 [ajp-bio-8009-exec-11051] ERROR org.intermine.api.profile.StorableBag  - Exception caught when closing Batch while addbagValues
java.sql.SQLException: Error writing to database, running statement COPY bagvalues (savedbagid, value, extra) FROM STDIN BINARY, data size = 7418
        at org.intermine.sql.writebatch.FlushJobPostgresCopyImpl.flush(FlushJobPostgresCopyImpl.java:56)
        at org.intermine.sql.writebatch.Batch$BatchFlusher.run(Batch.java:461)
        at java.lang.Thread.run(Thread.java:745)
Caused by: org.postgresql.util.PSQLException: ERROR: duplicate key value violates unique constraint "bagvalues_index1"
  Detail: Key (savedbagid, value, extra)=(59000002, A1AT_HUMAN, H. sapiens) already exists.
  Where: COPY bagvalues, line 1
        at org.postgresql.core.v3.QueryExecutorImpl.receiveErrorResponse(QueryExecutorImpl.java:2161)
        at org.postgresql.core.v3.QueryExecutorImpl.processCopyResults(QueryExecutorImpl.java:966)
        at org.postgresql.core.v3.QueryExecutorImpl.endCopy(QueryExecutorImpl.java:828)
        at org.postgresql.core.v3.CopyInImpl.endCopy(CopyInImpl.java:59)
        at org.postgresql.copy.CopyManager.copyIn(CopyManager.java:181)
        at org.postgresql.copy.CopyManager.copyIn(CopyManager.java:161)
        at org.intermine.sql.writebatch.FlushJobPostgresCopyImpl.flush(FlushJobPostgresCopyImpl.java:51)
        ... 2 more

```

**Comment by julie-sullivan** [comment]
I did recently change where the query.xsd is.

```
110816920 [ajp-bio-8009-exec-11051] INFO org.intermine.webservice.server.query.result.XMLValidator  - Using the xmlSchemaUrl http://www.humanmine.org/humanmine/webservice/query.xsd
110820447 [ajp-bio-8009-exec-11044] ERROR org.intermine.web.struts.InterMineExceptionHandler  - java.lang.NullPointerException: Path: "Protein.primaryIdentifier", pathToIndex: "{}", prefix: "Protein", query: "<query name="" model="genomi
c" view="Protein.primaryIdentifier Protein.primaryAccession Protein.organism.name Protein.length" longDescription=""><constraint path="Protein" op="IN" value="InterMine_workshop_ListA"/></query>"
        at org.intermine.api.results.WebResults.translateRow(WebResults.java:396)
        at org.intermine.api.results.WebResults.getResultElements(WebResults.java:375)
        at org.intermine.web.logic.results.PagedTable.updateRows(PagedTable.java:773)
        at org.intermine.web.logic.results.PagedTable.setPageAndPageSize(PagedTable.java:252)
        at org.intermine.web.struts.BagDetailsController.execute(BagDetailsController.java:240)
        at org.apache.struts.tiles.actions.TilesAction.execute(TilesAction.java:76)
        at org.apache.struts.action.RequestProcessor.processActionPerform(RequestProcessor.java:425)
```

What exactly is returning null?

**Comment by danielabutano** [comment]
it looks like the pathToIndex in WebResults is empty

* in a working list:
Path: "Gene.secondaryIdentifier",
**pathToIndex: "{Gene.primaryIdentifier=0, Gene=0, Gene.organism=1, Gene.symbol=0, Gene.secondaryIdentifier=0, Gene.organism.name=1}",**
prefix: "Gene",
query: "<query name="" model="genomic" view="Gene.secondaryIdentifier Gene.symbol Gene.primaryIdentifier Gene.organism.name" longDescription=""><constraint path="Gene" op="IN" value="five-genes"/></query>"

* in a similar not working list:
Path: "Gene.secondaryIdentifier",
**pathToIndex: "{}",**
prefix: "Gene",
query: "<query name="" model="genomic" view="Gene.secondaryIdentifier Gene.symbol Gene.primaryIdentifier Gene.organism.name" longDescription=""><constraint path="Gene" op="IN" value="FlyMine_Alzheimers-Use-Case"/></query>"

**Comment by danielabutano** [comment]
the other error ERROR org.intermine.api.profile.StorableBag  - Exception committing bagValues for bag: 59000002
java.sql.SQLException: Error writing to database, running statement COPY bagvalues
is related to the an other issue

**Comment by julie-sullivan** [comment]
Fixed by #1453
